bandit/src/scripts/backfill.ts

```diff
@@ -0,0 +1,65 @@
+import * as dateFns from 'date-fns';
+import { run as runCalculate } from "../calculate-lambda/calculate-lambda";
+import { run as runQuery } from "../query-lambda/query-lambda";
+
+/**
+ * Runs the query and creates a Dynamodb row for every hour from START to now.
+ * E.g.
+ * START="2024-03-28T01:00:00.000Z" TEST_NAME="2024-03-05_EPIC_PRIMARY__US" yarn backfill
+ */
+
+
+const wait = () =>
+	new Promise((resolve) => {
+		setTimeout(resolve, 2000);
+	});
+
+const retry = (fn: () => Promise<void>, retries = 5): Promise<void> => {
+	return fn().catch((err) => {
+		console.log('Retrying...');
+		if (retries > 0) {
+			return wait().then(() => retry(fn, retries - 1));
+		}
+		throw err;
+	});
+}
+
+const testName = process.env['TEST_NAME'] as string;
+const start = process.env['START'] as string;
+if (!testName || !start) {
+	console.error('Required environment variables: TEST_NAME or START');
+	process.exit(1);
+}
+const end = dateFns.set(new Date(), { minutes: 0, seconds: 0, milliseconds: 0 });
+
+const dates: Date[] = [];
+let date = new Date(start);
+while (date < end) {
+	date = dateFns.addHours(date, 1);
+	dates.push(date);
+}
+
+const tests = [{ name: testName }];
+
+const result = dates.reduce(
```

Comment:
the `reduce` here runs them in sequence, so the script will be slow if `START` was a long time ago

Reply:
just for my understanding - is the purpose of the `reduce` to chain the promises? what happens if we run them out of sequence? eg `Promise.all(dates.map( _async stuff_`

Reply:
good question, I guess we could end up with a lot of athena and dynamodb api calls being made at the same time, though this may not be an issue...